mod-source-record-manager-server/src/main/java/org/folio/services/afterprocessing/InstanceProcessingServiceImpl.java

```diff
@@ -69,20 +69,23 @@
   private MappingParametersProvider mappingParametersProvider;
   private MappingRuleService mappingRuleService;
   private JournalService journalService;
+  private HrIdFieldService hrIdFieldService;
 
   public InstanceProcessingServiceImpl(@Autowired JobExecutionSourceChunkDao jobExecutionSourceChunkDao,
                                        @Autowired MappingParametersProvider mappingParametersProvider,
                                        @Autowired MappingRuleService mappingRuleService,
+                                       @Autowired HrIdFieldService hrIdFieldService,
                                        @Autowired Vertx vertx) {
     this.jobExecutionSourceChunkDao = jobExecutionSourceChunkDao;
     this.mappingParametersProvider = mappingParametersProvider;
     this.mappingRuleService = mappingRuleService;
     this.journalService = JournalService.createProxy(vertx);
+    this.hrIdFieldService = hrIdFieldService;
   }
 
   @Override
-  public Future<Void> process(List<Record> records, String sourceChunkId, OkapiConnectionParams okapiParams) {
-    Future future = Future.future();
+  public Promise<Void> process(List<Record> records, String sourceChunkId, OkapiConnectionParams okapiParams) {
```

Comment:
please change the method signature to return Future object instead of a Promise object